Mage.Sets/src/mage/cards/c/ConspiracyTheorist.java

```diff
@@ -0,0 +1,178 @@
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.AttacksTriggeredAbility;
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.CostsImpl;
+import mage.abilities.costs.common.DiscardCardCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DoIfCostPaid;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.CardsImpl;
+import mage.constants.*;
+import mage.filter.FilterCard;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.util.CardUtil;
+import mage.watchers.Watcher;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class ConspiracyTheorist extends CardImpl {
+
+    public ConspiracyTheorist(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{R}");
+        
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SHAMAN);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Whenever Conspiracy Theorist attacks, you may pay {1} and discard a card. If you do, draw a card.
+        CostsImpl<Cost> costs = new CostsImpl<>();
+        costs.add(new ManaCostsImpl<>("{1}"));
+        costs.add(new DiscardCardCost());
+        this.addAbility(new AttacksTriggeredAbility(new DoIfCostPaid(new DrawCardSourceControllerEffect(1), costs).setText("you may pay {1} and discard a card. If you do, draw a card"), false));
+
+        // Whenever you discard one or more nonland cards, you may exile one of them from your graveyard. If you do, you may cast it this turn.
+        this.addAbility(new ConspiracyTheoristAbility(), new ConspiracyTheoristWatcher());
+    }
+
+    private ConspiracyTheorist(final ConspiracyTheorist card) {
+        super(card);
+    }
+
+    @Override
+    public ConspiracyTheorist copy() {
+        return new ConspiracyTheorist(this);
+    }
+}
+
+class ConspiracyTheoristAbility extends TriggeredAbilityImpl {
+
+    ConspiracyTheoristAbility() {
+        super(Zone.BATTLEFIELD, null);
+    }
+
+    ConspiracyTheoristAbility(ConspiracyTheoristAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DISCARDED_CARDS;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (event.getPlayerId().equals(getControllerId())) {
+            ConspiracyTheoristWatcher watcher = game.getState().getWatcher(ConspiracyTheoristWatcher.class);
+            if (watcher != null) {
```

Comment:
I think you're better off dropping the watcher and instead having the `DISCARDED_CARDS` event keep track of what was discarded

Reply:
fixed